### ISSUE TITLE
No arg generic methods fails code generation

### ISSUE THREAD

**Issue by yhrn** [immutables/immutables#1088]
If an abstract value class has a generic regular method (i.e. it's not an attribute accossor according to the definition [here](https://immutables.github.io/immutable.html#attributes)) that has a non-void return type and no arguments, Immutables will fail with the message: `Method 'methodName' cannot have own generic type parameters. Attribute accessors can only use enclosing type's type variables`

Below is an example of a (stupid) class that can be used to reproduce the problem on Immutables 2.7.5 (same result using 2.8.0-CriteriaRc0):

```java
package com.company.useless.utils

import java.util.ArrayList;
import java.util.List;
import org.immutables.value.Value;

@Value.Immutable
public abstract class ListFactory {
  abstract int initialCapacity();

  public <T> List<T> create() {
    return new ArrayList<>(initialCapacity());
  }
}
```

**Comment by elucash** [comment]
thank you for reporting! should be a bug

**Comment by elucash** [comment]
Looks like a good fix #1088 . I'll use compilation test from the example provided
```java
import java.util.ArrayList;
import java.util.List;
import org.immutables.value.Value;

@Value.Immutable
public abstract class ListFactory {
  abstract int initialCapacity();

  public <T> List<T> create() {
    return new ArrayList<>(initialCapacity());
  }
}
```

**Comment by elucash** [comment]
fixed in #8ce5911, I'll just see if there's a test
